infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogDataStore.java

```diff
@@ -181,65 +180,34 @@ void resetGlobalCompactionMark() {
     }
 
     /**
-     * Set the compacted addresses bitmap.
-     *
-     * @param newBitmap new compacted addresses bitmap.
+     * Return the current committed log tail.
      */
-    void updateCompactedAddresses(long segment, Roaring64NavigableMap newBitmap) {
-        Roaring64NavigableMap currBitmap = getCompactedAddresses(segment);
-        newBitmap.or(currBitmap);
-
-        newBitmap.runOptimize();
-        ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
-        DataOutput output = new DataOutputStream(byteStream);
-
-        try {
-            newBitmap.serialize(output);
-            ByteBuffer buffer = ByteBuffer.wrap(byteStream.toByteArray());
-            dataStore.put(getCompactedAddressesRecord(segment), buffer);
-        } catch (IOException ioe) {
-            throw new RuntimeException("Exception when attempting to " +
-                    "serialize compacted addresses bitmap", ioe);
-        } finally {
-            IOUtils.closeQuietly(byteStream);
-        }
+    long getCommittedTail() {
+        return committedTail.get();
     }
 
     /**
-     * Get the current compacted addresses bitmap.
-     * Caching is done on upper layer.
+     * Update the current committed log tail if the provided tail is greater.
+     *
+     * @param newCommittedTail a new address to update current global committed tail
      */
-    Roaring64NavigableMap getCompactedAddresses(long segment) {
-        ByteBuffer buf = dataStore.get(getCompactedAddressesRecord(segment));
-        if (buf == null) {
-            return new Roaring64NavigableMap();
-        }
-
-        // Cannot use flip() in case the buf is cached since flip() would
-        // change limit to position, but position is 0 if cached.
-        buf.rewind();
-        ByteArrayInputStream byteStream = new ByteArrayInputStream(
-                buf.array(), buf.position(), buf.remaining());
-        DataInput input = new DataInputStream(byteStream);
-
-        try {
-            Roaring64NavigableMap bitmap = new Roaring64NavigableMap();
-            bitmap.deserialize(input);
-            return bitmap;
-        } catch (IOException ioe) {
-            throw new SerializerException("Exception when attempting to " +
-                    "deserialize compacted addresses bitmap.", ioe);
-        } finally {
-            IOUtils.closeQuietly(byteStream);
+    void updateCommittedTail(long newCommittedTail) {
+        if (updateIfGreater(committedTail, newCommittedTail, COMMITTED_TAIL_RECORD)) {
+            log.debug("Updated committed log tail to: {}", newCommittedTail);
+        } else {
+            log.trace("New committed log tail not greater than current, ignore.");
```

Comment:
Is it better to log the concrete value of the newCommittedTail even if it is not greater than existing one?

Reply:
Done.